### PR TITLE
Remove attribute file when it is not longer used

### PR DESCRIPTION
**Pull request by IKarbowiak**
- Clean `file_url` before saving as `AttributeValue` - fix removing file when attribute value is removed
- Add hash to file name in `FileUpload` mutation
- Remove corresponding `AttributeValue`s of type `File`/`Reference`/`Rich text`/`Numeric`, when `Product`/`ProductVariant`/`Page/ProductType/PageType` is removed
- Save files uploaded by `FileUpload` mutation to `file_upload` directory

Linked tasks: [SALEOR-3083](https://app.clickup.com/t/2549495/SALEOR-3083) and [SALEOR-3079](https://app.clickup.com/t/2549495/SALEOR-3079)

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3083 FileAttributes are not removed even if page/product/attribute they were assigned to has been removed](https://app.clickup.com/t/2549495/SALEOR-3083)

**Review by eCzerniak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

File: saleor/attribute/__init__.py [hunk 1]
```diff
@@ -27,6 +27,14 @@ class AttributeInputType:
         MULTISELECT,
     ]
 
+    # list of the input types that are unique per instances
+    TYPES_WITH_UNIQUE_VALUES = [
```
Comment (maarcingebala) [inline saleor/attribute/__init__.py:31]: Types with unique values are those that are not in `TYPES_WITH_CHOICES` I guess? We could use it and inverse the if-condition.

**Review by maarcingebala — COMMENTED** [review]
Product and variants are removed from Saleor also when removing a product type; the same goes for pages and page types. We should call `delete_assigned_attribute_values` in these cases as well.

**Review by eCzerniak — APPROVED** [review]

File: saleor/attribute/__init__.py (hunk 1 above)
Comment (IKarbowiak) [reply saleor/attribute/__init__.py:31]: We agreed to leave it as it is.

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**